awswrangler/data_api/redshift.py

```diff
@@ -72,7 +94,18 @@ def _validate_auth_method(self) -> None:
         if self.workgroup_name == "" and self.secret_arn == "" and self.db_user == "":
             raise ValueError("Either `secret_arn` or `db_user` must be set for authentication")
 
-    def _execute_statement(self, sql: str, database: Optional[str] = None) -> str:
+    def _execute_statement(
+        self,
+        sql: str,
+        database: Optional[str] = None,
+        transaction_id: Optional[str] = None,
+        parameters: Optional[List[Dict[str, Any]]] = None,
+    ) -> str:
+        if transaction_id:
```

Comment:
Since only rds._execute_statement is implementing transaction_id and parameters, perhaps we should only define it there and not in the underlying abstract method?

Reply:
MyPy complains about the method signatures not matching, so I added it here. I could remove them as you suggess, but I'll need a type ignore. I feel like this style with an exception is the more robust way to go.